src/main.ts

```diff
@@ -10,12 +10,19 @@ async function run() {
         const output: string[] = [];
         const token = core.getInput('token', {required: false});
         const filtersInput = core.getInput('filters', {required: true});
-        const filters = filtersInput.split('\n').map(s => s.trim()).filter(s => s.length > 0);
+        const filters = filtersInput.split('\n').map(s => s.trim()).filter(s => s.length > 0).
+            map(s => {
+                const parts = s.split(':', 2)
+                const idx = parts.length == 2
+                    ? parseInt(parts[1], 10)
```

Comment:
I think we should ensure that we have receive a positive number by wrapping the parseInt with `Math.min(parseInt(...), 0)`

Also I think, for the sake of readibility, we should decrease the result of `parseInt` by 1 so that:

```yaml
 with:
    filters: |
      apps/*/**:2    # Indicate that we want to use 'apps/*/' as base folder (therefore idx should be 1)
      apps/*/**:1    # Indicate that we only want to use 'apps/' as base folder (therefore idx should be 0)
```

Reply:
It should be `Math.max`. ;)

Reply:
Indeed :smile: